sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeq.scala

```diff
@@ -71,7 +71,10 @@ object OffsetSeq {
  * @param batchTimestampMs: The current batch processing timestamp.
  * Time unit: milliseconds
  */
-case class OffsetSeqMetadata(var batchWatermarkMs: Long = 0, var batchTimestampMs: Long = 0) {
+case class OffsetSeqMetadata(
+    var batchWatermarkMs: Long = 0,
+    var batchTimestampMs: Long = 0,
+    var numShufflePartitions: Int = 0) {
```

Comment:
It's better to use `conf: Map[String, String]` here because we probably will add more confs to this class in future.